src/plugins/score-plugin-avnd/Spat/Rotator.hpp

```diff
@@ -0,0 +1,152 @@
+#pragma once
+#include <iostream>
+//#include <vector>
+//#include <iomanip>
+#include <halp/audio.hpp>
+#include <halp/controls.hpp>
+#include <halp/meta.hpp>
+//#include <experimental/mdspan>
+
+#include <saf.h>
+//#include <rotator.h>
+//#include <cblas.h>
+
+#define MAX_ORDER 7
+#define MAX_NSH ((MAX_ORDER + 1) * (MAX_ORDER + 1))
+
+class Rotator
+{
+public:
+  halp_meta(name, "Rotator") halp_meta(c_name, "avnd_rotator")
+      halp_meta(uuid, "82bdb9f5-9cf8-440e-8675-c0caf4fc59b9")
+
+          using setup = halp::setup;
+  using tick = halp::tick;
+
+  struct
+  {
+    halp::dynamic_audio_bus<"Input", float> audio;
+    halp::hslider_i32<
+        "Order",
+        halp::range{.min = 0, .max = MAX_ORDER, .init = 0}>
+        order;
+    halp::knob_f32<"Yaw", halp::range{.min = -180.0, .max = 180.0, .init = 0}>
+        yaw;
+    halp::
+        knob_f32<"Pitch", halp::range{.min = -180.0, .max = 180.0, .init = 0}>
+            pitch;
+    halp::knob_f32<"Roll", halp::range{.min = -180.0, .max = 180.0, .init = 0}>
+        roll;
+  } inputs;
+
+  struct
+  {
+    halp::dynamic_audio_bus<"Output", float> audio;
+  } outputs;
+
+  /*Rotator()
+    {
+
+    }
+
+    ~Rotator()
+    {
+
+    }*/
+
+  halp::setup setup_info;
+  void prepare(halp::setup info)
+  {
+    previous_values.resize(info.input_channels);
+  }
+
+  void operator()(halp::tick t)
+  {
+    if (inputs.audio.channels == 0)
+      return;
+
+    order = inputs.order;
+    nSH = (order + 1) * (order + 1);
+    nSamples = t.frames;
+
+    yaw = inputs.yaw * M_PI / 180.0;
+    pitch = inputs.pitch * M_PI / 180.0;
+    roll = inputs.roll * M_PI / 180.0;
+
+    /*while(nSH > inputs.audio.channels)
+        {
+            order -= 1;
+            nSH = (order+1)*(order+1);
+        }*/
+
+    float** in = inputs.audio.samples;
+    float** out = outputs.audio.samples;
+
+    float inFrame[MAX_NSH][nSamples];
```

Comment:
Please use a std::vector<float> member, resized in the `prepare` method instead (e.g. `m_inFrame.resize(MAX_NSH * nSamples);`) : it's not legal C++ to have arrays of dynamic size sadly (and some compilers like on windows do not accept it.